src/renderers/WebGLRenderer.js

```diff
@@ -2268,6 +2268,22 @@ class WebGLRenderer {
 					// Color and depth texture must be rebound in order for the swapchain to update.
 					textures.rebindTextures( renderTarget, properties.get( renderTarget.texture ).__webglTexture, properties.get( renderTarget.depthTexture ).__webglTexture );
 
+				} else if ( renderTarget.depthBuffer || renderTarget.depthTexture ) {
```

Comment:
I've tested the PR with [webgl_rtt](https://threejs.org/examples/webgl_rtt) and it does not look right to me that this code path is taken in normal `setRenderTarget()` calls. The additional bindings executed in `setupDepthRenderbuffer()` are wasted for all scenarios with a static depth buffer.

Isn't possible to make the exchange more explicit? E.g. use a flag that is set to true when depth texture reference is changed?

Reply:
Good call - I just added a check in `setupDepthRenderbuffer` to avoid rebinding the depth buffer if it has already been bound. Let me know what you think.

Reply:
cc @Mugen87 - just wanted to check on this again. The PR has now been changed so that the depth buffer is only updated and rebound if it has changed.